pom.xml

```diff
@@ -92,6 +128,59 @@
                     </execution>
                 </executions>
             </plugin>
+
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+                <version>0.8.4</version>
+                <executions>
+                    <execution>
+                        <goals>
```

Comment:
I saw that you added an on the test phase the id `<id>post-unit-test</id>`, we can add an id to this execution as well. Something like `<id>pre-unit-test</id>` would do the trick. 

Also please consider renaming the id of the checkstyle and pmd executions as they are both `validate`.  Maybe `<id>checkstyle-validate</id>` and <id>pmd-validate</id>.

It seems these id's are printed in the console when ran.
Example: 
```
[INFO] --- jacoco-maven-plugin:0.8.4:prepare-agent (pre-unit-test) @ imen ---
```